Orange/widgets/data/owselectcolumns.py

```diff
@@ -605,7 +609,8 @@ def update_completer_model(self, *_):
                         [list(v.attributes.items()) for v in vars], [])
         items.extend(["%s=%s" % item for item in labels])
         items.extend(reduce(list.__add__, list(map(list, labels)), []))
-
+        if self.auto_commit:
```

Comment:
@rokgomiscek This is wrong: you have to call `self.commit`. `gui.auto_commit` replaces your `self.commit` with a function the checks the auto commit flag and calls the original commit or sets the dirty flag.

Therefore, remove `if` and always call.